File: network/src/main/java/com/difft/android/websocket/internal/push/exceptions/MismatchedDevicesException.java

package com.difft.android.websocket.internal.push.exceptions;

import com.difft.android.websocket.api.push.exceptions.NonSuccessfulResponseCodeException;
import com.difft.android.websocket.internal.push.MismatchedDevices;

public class MismatchedDevicesException extends NonSuccessfulResponseCodeException {

  private final MismatchedDevices mismatchedDevices;

  public MismatchedDevicesException(MismatchedDevices mismatchedDevices) {
    super(409);
    this.mismatchedDevices = mismatchedDevices;
  }

  public MismatchedDevices getMismatchedDevices() {
    return mismatchedDevices;
  }
}

File: network/src/main/java/com/difft/android/websocket/internal/push/exceptions/MismatchedDevicesException.kt
package com.difft.android.websocket.internal.push.exceptions

import com.difft.android.websocket.api.push.exceptions.NonSuccessfulResponseCodeException
import com.difft.android.websocket.internal.push.MismatchedDevices

class MismatchedDevicesException(
    val mismatchedDevices: MismatchedDevices
) : NonSuccessfulResponseCodeException(409)
